controller/src/main/java/io/pravega/controller/server/rpc/auth/AuthHelper.java

```diff
@@ -31,22 +31,29 @@ public static AuthHelper getDisabledAuthHelper() {
         return new AuthHelper(false, "");
     }
 
-    public String checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) {
+    public boolean isAuthorized(String resource, AuthHandler.Permissions permission) {
         if (isAuthEnabled) {
             PravegaInterceptor currentInterceptor = PravegaInterceptor.INTERCEPTOR_OBJECT.get();
 
             AuthHandler.Permissions allowedLevel;
             if (currentInterceptor == null) {
-                //No interceptor, and authorization is enabled. Means no access is granted.
+                //No interceptor, and authorization is enabled. That means no access is granted.
                 allowedLevel = AuthHandler.Permissions.NONE;
             } else {
                 allowedLevel = currentInterceptor.authorize(resource);
             }
-            if (allowedLevel.ordinal() < expectedLevel.ordinal()) {
-                throw new RuntimeException(new AuthorizationException("Access not allowed"));
-            }
+            return (allowedLevel.ordinal() < permission.ordinal()) ? false : true;
+        } else {
+            return true;
+        }
+    }
+
+    public String checkAuthorization(String resource, AuthHandler.Permissions expectedLevel) {
```

Comment:
`checkAuthorization` was an existing method used in a lot of places. `isAuthorized` is a new method I added for list operations for checking authorization for each scope/stream without having to depend on occurrence of an exception to determine failed authorization. I moved some of the logic inside `checkAuthorization` into the new `isAuthorized` method. I didn't want to modify existing code that depends on checkAuthorization, to avoid changing any semantics in the authorization behavior, and also because that change'd have been irrelevant to problem at hand.